src/zui/ZUIEditableImage.tsx

```diff
@@ -1,47 +1,98 @@
-import { Box, IconButton } from '@mui/material';
+import { useState } from 'react';
+import { Box, IconButton, useTheme } from '@mui/material';
 import { Clear, Edit } from '@mui/icons-material';
 import Image, { ImageProps } from 'next/image';
 
+import FileLibraryDialog from 'features/files/components/FileLibraryDialog';
+import messageIds from './l10n/messageIds';
+import { Msg } from 'core/i18n';
+import { useNumericRouteParams } from 'core/hooks';
+import { ZetkinFile } from 'utils/types/zetkin';
+
 interface ZUIEditableImageProps {
   alt: string;
-  src: string;
-  onEdit: () => void;
-  onReset: () => void;
+  file: ZetkinFile | null;
+  onFileSelect: (file: ZetkinFile | null) => void;
 }
 
-const ZUIEditableImage: React.FC<ZUIEditableImageProps & ImageProps> = ({
-  alt,
-  onEdit,
-  onReset,
-  src,
-  ...imageProps
-}) => {
+const ZUIEditableImage: React.FC<
+  ZUIEditableImageProps & Omit<ImageProps, 'src'>
+> = ({ alt, file, onFileSelect, ...imageProps }) => {
+  const theme = useTheme();
+  const { orgId } = useNumericRouteParams();
+  const [selecting, setSelecting] = useState(false);
+
   return (
-    <Box style={{ position: 'relative' }}>
+    <Box
+      onClick={(ev) => {
+        ev.stopPropagation();
+      }}
+      style={{ position: 'relative' }}
+    >
       <Box style={{ bottom: 10, position: 'absolute', right: 10, zIndex: 1 }}>
-        <IconButton
-          data-testid="ZetkinEditableImage-resetButton"
-          onClick={() => onReset()}
-          size="large"
-          style={{
-            backgroundColor: 'rgba(255,255,255,0.7)',
-            marginRight: 10,
-          }}
-        >
-          <Clear />
-        </IconButton>
-        <IconButton
-          data-testid="ZetkinEditableImage-editButton"
-          onClick={() => onEdit()}
-          size="large"
-          style={{
-            backgroundColor: 'rgba(255,255,255,0.7)',
+        {file && (
+          <>
+            <IconButton
+              data-testid="ZetkinEditableImage-resetButton"
+              onClick={() => onFileSelect(null)}
+              size="large"
+              style={{
+                backgroundColor: 'rgba(255,255,255,0.7)',
+                marginRight: 10,
+              }}
+            >
+              <Clear />
+            </IconButton>
+            <IconButton
+              data-testid="ZetkinEditableImage-editButton"
+              onClick={() => setSelecting(true)}
+              size="large"
+              style={{
+                backgroundColor: 'rgba(255,255,255,0.7)',
```

Comment:
Is there a specific reason for using a non-theme color here, when having the theme in the file?

Reply:
I'm not sure. I did not write this code (now at least) but only moved it from the `TaskPreviewSection` component. But I'm guessing the reason is that it's a semi-transparent color, and I'm not sure those exist in the theme?